include/gridtools/stencil_composition/structured_grids/backend_mc/iterate_domain_mc.hpp

```diff
@@ -42,106 +39,77 @@ namespace gridtools {
             return value;
         }
 
-        /**
-         * @brief compute thread offsets for temporaries
-         */
         template <class LocalDomain>
-        struct set_offset_for_temporaries_f {
+        struct set_base_offset_f {
             LocalDomain const &m_local_domain;
+            int_t m_i_block_base;
+            int_t m_j_block_base;
             typename LocalDomain::ptr_map_t &m_dst;
 
-            template <class Arg>
+            template <class Arg, enable_if_t<is_tmp_arg<Arg>::value, int> = 0>
             GT_FORCE_INLINE void operator()() const {
-
-                using strides_kind_t = GT_META_CALL(strides_kind_from_arg, (LocalDomain, Arg));
-
+                using sid_t = GT_META_CALL(storage_from_arg, (LocalDomain, Arg));
+                using strides_kind_t = GT_META_CALL(sid::strides_kind, sid_t);
                 auto length = at_key<strides_kind_t>(m_local_domain.m_total_length_map);
-                int_t offset = std::lround(length * thread_factor());
+                GT_META_CALL(sid::ptr_diff_type, sid_t) offset = std::lround(length * thread_factor());
                 assert(offset == ((long long)length * omp_get_thread_num()) / omp_get_max_threads());
+                auto const &strides = at_key<strides_kind_t>(m_local_domain.m_strides_map);
+                GT_STATIC_ASSERT(is_storage_info<strides_kind_t>::value, GT_INTERNAL_ERROR);
+                sid::shift(offset, sid::get_stride<dim::i>(strides), strides_kind_t::halo_t::template at<0>());
+                sid::shift(offset, sid::get_stride<dim::j>(strides), strides_kind_t::halo_t::template at<1>());
                 at_key<Arg>(m_dst) += offset;
             }
+
+            template <class Arg, enable_if_t<!is_tmp_arg<Arg>::value, int> = 0>
+            GT_FORCE_INLINE void operator()() const {
+                using sid_t = GT_META_CALL(storage_from_arg, (LocalDomain, Arg));
+                using strides_kind_t = GT_META_CALL(sid::strides_kind, sid_t);
+                auto &ptr = at_key<Arg>(m_dst);
+                auto const &strides = at_key<strides_kind_t>(m_local_domain.m_strides_map);
+                sid::shift(ptr, sid::get_stride<dim::i>(strides), m_i_block_base);
+                sid::shift(ptr, sid::get_stride<dim::j>(strides), m_j_block_base);
+            }
         };
 
+#ifdef __SSE__
+        template <class T, class PtrDiff, class Strides>
+        GT_FORCE_INLINE void do_prefetch(T *ptr, PtrDiff offset, Strides const &strides, int_t dist) {
```

Comment:
This is pretty useless now as it was only used in the disabled code further below that you deleted (inner k fusion with max_extent == 0), so I think you can safely remove all prefetching stuff.

Reply:
done